src/test/java/com/microsoft/sqlserver/jdbc/unit/serial/DTOSerialTest.java

```diff
@@ -0,0 +1,199 @@
+package com.microsoft.sqlserver.jdbc.unit.serial;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.SQLServerException;
+import com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement;
+import com.microsoft.sqlserver.jdbc.SQLServerResultSet;
+import com.microsoft.sqlserver.jdbc.TestUtils;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+
+import microsoft.sql.DateTimeOffset;
+
+
+@RunWith(JUnitPlatform.class)
+public class DTOSerialTest extends AbstractTest {
+    private static final SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss.SSSSS XXX");
+    private static String dateString;
+
+    @Test
+    public void testDSerial() throws Exception {
+        sdf.setTimeZone(TimeZone.getTimeZone("Z"));
+        dateString = sdf.format(new Date());
+
+        try (Connection conn = DriverManager.getConnection(connectionString);
+                Statement stmt = conn.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE)) {
+
+            // create a DTO
+            ResultSet rs = stmt.executeQuery(
+                    "SELECT CAST('" + dateString + "' AS datetimeoffset(7)) AS" + "   'datetimeoffset IS08601' ");
+            rs.next();
+            verifyCorrectSerialization(((SQLServerResultSet) rs).getDateTimeOffset(1));
+            verifyMessedSerialization();
+        }
+    }
+
+    @Test
+    public void testESerial() throws Exception {
+        try (Connection conn = DriverManager.getConnection(connectionString);
+                Statement stmt = conn.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE)) {
+
+            // raise an error.
+            SQLServerException currException = null;
+
+            try {
+                stmt.executeUpdate("RAISERROR ('foo', 13,1) WITH LOG");
+
+            } catch (SQLServerException x) {
+                currException = x;
+            }
+
+            // store the info
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            ObjectOutputStream out = new ObjectOutputStream(bos);

```

Comment:
Streams should be created in try-with-resources block so they are closed in the end.